tests/unit/test_protocol.py

```diff
@@ -28,6 +29,86 @@ def test_column_null():
+        b'\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x00'
+    assert protocol.build_numeric(Decimal('-0')) == \
+        b'\x00\x00\x00\x08\x00\x00\x00\x00@\x00\x00\x00'
+
+    assert protocol.build_numeric(Decimal('5.7288')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x04\x00\x05\x1cx'
+    assert protocol.build_numeric(Decimal('2.2622600000000003164')) == \
+        b'\x00\x00\x00\x14\x00\x06\x00\x00\x00\x00\x00\x13\x00\x02\n>\x17p\x00\x00\x00\x03\x06h'
+    assert protocol.build_numeric(Decimal('5.69483')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00\x05\x1b$\x0b\xb8'
+    assert protocol.build_numeric(Decimal('152.080')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x03\x00\x98\x03 '
+    assert protocol.build_numeric(Decimal('44.80428')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00,\x1fj\x1f@'
+    assert protocol.build_numeric(Decimal('2.53620')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x05\x00\x02\x14\xf2'
+    assert protocol.build_numeric(Decimal('6.4965')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x04\x00\x06\x13e'
+    assert protocol.build_numeric(Decimal('71.78358')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00G\x1e\x9b\x1f@'
+    assert protocol.build_numeric(Decimal('7.4625')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x04\x00\x07\x12\x11'
+    assert protocol.build_numeric(Decimal('38.69635')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00&\x1b3\x13\x88'
+    assert protocol.build_numeric(Decimal('5.7288')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x04\x00\x05\x1cx'
+    assert protocol.build_numeric(Decimal('2.2622600000000003164')) == \
+        b'\x00\x00\x00\x14\x00\x06\x00\x00\x00\x00\x00\x13\x00\x02\n>\x17p\x00\x00\x00\x03\x06h'
+    assert protocol.build_numeric(Decimal('5.69483')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00\x05\x1b$\x0b\xb8'
+    assert protocol.build_numeric(Decimal('152.080')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x03\x00\x98\x03 '
+    assert protocol.build_numeric(Decimal('44.80428')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00,\x1fj\x1f@'
+    assert protocol.build_numeric(Decimal('2.53620')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x05\x00\x02\x14\xf2'
+    assert protocol.build_numeric(Decimal('6.4965')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x04\x00\x06\x13e'
+    assert protocol.build_numeric(Decimal('71.78358')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00G\x1e\x9b\x1f@'
+    assert protocol.build_numeric(Decimal('7.4625')) == \
+        b'\x00\x00\x00\x0c\x00\x02\x00\x00\x00\x00\x00\x04\x00\x07\x12\x11'
+    assert protocol.build_numeric(Decimal('38.69635')) == \
+        b'\x00\x00\x00\x0e\x00\x03\x00\x00\x00\x00\x00\x05\x00&\x1b3\x13\x88'
```

Comment:
it would be much easier to read those values if they were converted to `.hex()` representation. Now I don't even know what is the actual result of this function.

Reply:
Done